server/src/main/java/org/opensearch/index/shard/IndexShard.java

```diff
@@ -3669,22 +3662,20 @@ public synchronized void onNewCheckpoint(
             return;
         }
         if (isReplicating()) {
-            logger.info("Ignore - shard is currently replicating to a checkpoint");
+            logger.debug("Ignore - shard is currently replicating to a checkpoint");
             return;
         }
         try {
             markAsReplicating();
             final ReplicationCheckpoint checkpoint = request.getCheckpoint();
             logger.trace("Received new checkpoint {}", checkpoint);
-            // TODO: segrep - these are the states set after we perform our initial store recovery.
             segmentReplicationReplicaService.startReplication(
                 checkpoint,
                 this,
                 source,
                 new SegmentReplicationReplicaService.SegmentReplicationListener() {
                     @Override
                     public void onReplicationDone(SegmentReplicationState state) {
-                        markReplicationComplete();
```

Comment:
I moved this to when ReplicationTarget's finalize method completes because I didn't like SegmentReplicationListener being responsible for marking it as complete.  With that said I think we can do better by moving it to the onDone/onCancel/onFail methods of SegmentReplicationTarget.  Will also move markAsReplicating to ReplicationTarget as well so these classes don't have to manage that.

Reply:
cleaned this up in latest commit, what do you think?

Reply:
Wasn't the plan to move `markAsReplicating` to `ReplicationTarget` as well? Right now, that seems to be in `ReplicaService` so the lifecycle management is distributed across two classes, which smells. How do we improve this?

Reply:
Could you create a task to track it, so we don't forget?